src/util/ccl/rccl.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpiimpl.h"
+
+#ifdef ENABLE_RCCL
+
+#define HIP_ERR_CHECK(ret)             \
+    if (unlikely((ret) != hipSuccess)) \
+    goto fn_fail
+
+#define NCCL_ERR_CHECK(ret)             \
```

Comment:
I recommend calling this `RCCL_ERR_CHECK` so you do not get a re-definition error if you try to compile with both NCCL and RCCL, even though I recognize it is identical to `NCCL_ERR_CHECK` because RCCL uses NCCL values like `ncclSuccess`